### PR TITLE
[QC-769] add detector to proxy names and fix check name

### PR DESCRIPTION
**Pull request by Barthelemy**

### CODE REVIEW

**Comment by Barthelemy** [comment]
@TimoWilken o2-dataflow  is also failing without logs

File: Framework/test/testInfrastructureGenerator.cxx [hunk 1]
```diff
@@ -103,7 +103,8 @@ BOOST_AUTO_TEST_CASE(qc_factory_remote_test)
   auto skeletonTaskProxy = std::find_if(
     workflow.begin(), workflow.end(),
     [](const DataProcessorSpec& d) {
-      return d.name == "skeletonTask-proxy" &&
+      std::cout << "asdf : " << d.name << std::endl;
```
Comment (knopers8) [inline Framework/test/testInfrastructureGenerator.cxx:106]: could be removed

**Review by knopers8 — APPROVED** [review]
thanks, i have two pedantic comments, but feel free to address them later if you want to proceed with the release

File: Framework/include/QualityControl/InfrastructureGenerator.h [hunk 2]
```diff
@@ -186,15 +187,10 @@ class InfrastructureGenerator
                                                         const std::string& control);
   static void generateLocalTaskLocalProxy(framework::WorkflowSpec& workflow,
                                           size_t id,
-                                          std::string taskName,
-                                          std::string remoteHost,
-                                          std::string remotePort,
-                                          const std::string& control);
+                                          const TaskSpec& taskSpec);
```
[the remark this answers is not in the document]
Comment (Barthelemy) [reply Framework/include/QualityControl/InfrastructureGenerator.h:190]: agreed, fixing

**Pull request merged**